### DISCUSSION TITLE
IconLayer mask documentation

### DISCUSSION THREAD

**Discussion by am2222** [visgl/deck.gl#9063] [Q&A] [answered]
Hi,
Is there any documentations about how IconColor with `mask=true` works on Iconlayer? I have an icon which has a border and with mask=false it will not apply the iconColor (which is expected) but once I turn mask=true it applies color to entire icon and removes the border too. 
In general I want to see if we can apply icon color to a portion of the png. Does deckgl multiply color to the existing colors in icon?

**Comment by Pessimistress** [comment, marked as answer]
When an icon has `mask:true`, it only uses the source image as a transparency mask. More specifically,

`color = vec4(instanceColor.rgb, instanceColor.a * textureColor.a * opacity)`

You can subclass the IconLayer to apply your own masking logic. [example](https://codepen.io/Pessimistress/pen/QWXpyyL)
